lib/gen_socket_client.ex

```diff
@@ -71,7 +71,12 @@ defmodule Phoenix.Channels.GenSocketClient do
 
   @type socket_opts :: [{:serializer, module}]
   @type callback_state :: any
-  @type transport :: %{transport: pid | nil, message_refs: :ets.tab, serializer: module}
+  @opaque transport :: %{
+    transport_mod: module,
+    transport: pid | nil,
```

Comment:
It is in fact called `transport_pid`, which is IMO better than a vague `pid`. Typespec is wrong though. I'll fix it, and also amend previous short comments.